src/components/Table/index.tsx

```diff
@@ -0,0 +1,167 @@
+    background: ${({ theme }) => theme.surface3};
+  }
+`
+const StyledData = styled.td`
+  max-width: 160px;
+  padding: 2px 6px 2px 6px;
+  text-align: right;
+  position: relative;
+  width: fit-content;
+`
+
+interface TableProps<Data extends object> {
+  columns: Column<Data>[]
+  data: Data[]
+  smallHiddenColumns: IdType<Data>[]
+  mediumHiddenColumns: IdType<Data>[]
+  largeHiddenColumns: IdType<Data>[]
+  extraLargeHiddenColumns: IdType<Data>[]
+  dataTestId?: string
+}
+
+export function Table<Data extends object>({
+  columns,
+  data,
+  smallHiddenColumns,
+  mediumHiddenColumns,
+  largeHiddenColumns,
+  extraLargeHiddenColumns,
+  dataTestId,
+  ...props
+}: TableProps<Data>) {
+  const theme = useTheme()
+  const { width } = useWindowSize()
+  const { getTableProps, getTableBodyProps, headerGroups, rows, prepareRow, setHiddenColumns } = useTable(
+    {
+      columns,
+      data,
+      ...props,
+    },
+    useSortBy
+  )
+
+  useEffect(() => {
+    if (!width) return
+
+    if (width <= theme.breakpoint.sm) {
+      setHiddenColumns(smallHiddenColumns)
+    } else if (width <= theme.breakpoint.md) {
+      setHiddenColumns(mediumHiddenColumns)
+    } else if (width <= theme.breakpoint.lg) {
+      setHiddenColumns(largeHiddenColumns)
+    } else {
+      setHiddenColumns(extraLargeHiddenColumns)
+    }
+  }, [
+    width,
+    setHiddenColumns,
+    columns,
+    smallHiddenColumns,
+    mediumHiddenColumns,
+    largeHiddenColumns,
+    theme.breakpoint,
+    extraLargeHiddenColumns,
+  ])
+
+  return (
+    <Container>
+      <StyledTable {...getTableProps()} data-testid={dataTestId}>
+        <thead
+          style={{
+            position: 'sticky',
+            top: '0px',
+            zIndex: 1,
+            boxShadow: `inset 0 -2px 0 ${theme.surface3}`,
+            background: `${theme.surface1}`,
+          }}
+        >
+          {headerGroups.map((headerGroup, headerGroupIndex) => (
+            <tr {...headerGroup.getHeaderGroupProps()} key={`header-group-${headerGroupIndex}`}>
+              {headerGroup.headers.map((column, index) => {
+                return (
+                  <th {...column.getHeaderProps(column.getSortByToggleProps())} key={column.id}>
+                    <StyledHeaderContent leftMost={index === 0} disabled={column.disableSortBy}>
+                      <Box as="span" color="neutral2" position="relative">
+                        {column.isSorted ? (
+                          column.isSortedDesc ? (
+                            <ArrowChangeUp width="16px" height="16px" style={{ position: 'absolute', top: 3 }} />
+                          ) : (
+                            <ArrowChangeDown width="16px" height="16px" style={{ position: 'absolute', top: 3 }} />
+                          )
+                        ) : (
+                          ''
+                        )}
+                      </Box>
+                      <Box as="span" paddingLeft={column.isSorted ? '18' : '0'}>
```

Comment:
random padding left here feels hacky.. does sorting add an icon that you're accounting for here?

Reply:
Removed. Will be adding sorting into a follow-up PR.